tests/e2e/distribution.go

```diff
@@ -75,3 +76,118 @@ func (s *CCVTestSuite) TestRewardsDistribution() {
 	s.Require().Greater(ibcCoinIndex, -1)
 	s.Require().True(communityCoins[ibcCoinIndex].Amount.Equal(sdk.NewDecCoinFromCoin(providerExpectedRewards[0]).Amount))
 }
+
+// TestEndBlockRD tests that the last transmission block height (LTBH) is correctly updated after the expected
+// number of block have passed. It also checks that the IBC transfer transfer states are discarded if
+// the reward distribution to the provider has failed.
+func (s *CCVTestSuite) TestEndBlockRD() {
+
+	testCases := []struct {
+		name                    string
+		prepareRD               bool
+		corruptTransChannel     bool
+		expLBThUpdated          bool
+		expEscrowBalanceChanged bool
+	}{
+		{
+			name:                    "should not update LBTH before blocks per dist trans block are passed",
+			prepareRD:               false,
+			corruptTransChannel:     false,
+			expLBThUpdated:          false,
+			expEscrowBalanceChanged: false,
+		},
+		{
+			name:                    "should update LBTH when blocks per dist trans or more block are passed",
+			prepareRD:               true,
+			corruptTransChannel:     false,
+			expLBThUpdated:          true,
+			expEscrowBalanceChanged: true,
+		},
+		{
+			name:                    "should update LBTH and discard the IBC transfer states when sending rewards to provider fails",
+			prepareRD:               true,
+			corruptTransChannel:     true,
+			expLBThUpdated:          true,
+			expEscrowBalanceChanged: false,
+		},
+	}
+
+	for _, tc := range testCases {
+
+		s.SetupTest()
+
+		// ccv and transmission channels setup
+		s.SetupCCVChannel(s.path)
+		s.SetupTransferChannel()
+		bondAmt := sdk.NewInt(10000000)
+		delAddr := s.providerChain.SenderAccount.GetAddress()
+		delegate(s, delAddr, bondAmt)
+		s.providerChain.NextBlock()
+
+		// relay VSC packets from provider to consumer
+		relayAllCommittedPackets(s, s.providerChain, s.path, ccv.ProviderPortID, s.path.EndpointB.ChannelID, 1)
+
+		consumerKeeper := s.consumerApp.GetConsumerKeeper()
+		consumerBankKeeper := s.consumerApp.GetE2eBankKeeper()
+		ctx := s.consumerCtx()
```

Comment:
Note this is actually a cached context. That's why most tests use `consumerCtx()` inline

Reply:
@MSalopek, changing the test to use `consumerCtx()` inline fixed the test setup ordering weirdness we were seeing. 

`CommitNBlocks` was being called on a different ctx than the IBC keeper was being obtained from